.github/workflows/ci.yml

```diff
@@ -95,23 +97,45 @@ jobs:
       - name: Install Go
         uses: actions/setup-go@v5
         with:
-          go-version: 1.22.x
+          # Take Go version to install from go.mod.
+          go-version-file: 'go.mod'
+      - name: Set GOPATH
+        run: |
+          echo "GOPATH=$(go env GOPATH)" >> $GITHUB_ENV
+      - name: Copy repo to GOPATH
+        # This is needed as gomobile is still unaware of go modules, so the repo must be in GOPATH
+        run: |
+          mkdir -p $GOPATH/$(dirname $GO_SRC_DIR)
+          cp -a ${{github.workspace}} $GOPATH/$(dirname $GO_SRC_DIR)
       - name: Install Node.js
         uses: actions/setup-node@v4
         with:
           node-version: '20.x'
+      - name: Install Qt
+        run: |
+          brew install qt@5
+          echo "/opt/homebrew/opt/qt@5/bin" >> $GITHUB_PATH
       - name: Build macOS app
-        run: >
-          ./scripts/github-ci.sh qt-osx;
+        run: |
+          pushd $GOPATH/$GO_SRC_DIR
```

Comment:
no strong feelings, but I prefer `make -C <dir> target` or `(cd <dir>; make target)` over `pushd`/`popd`. so that you don't have to pop all the pushes you have done.

Reply:
fixed